src/VisualStudio/Core/Def/Implementation/VisualStudioWorkspaceContextService.cs

```diff
@@ -31,7 +42,9 @@ public bool IsCloudEnvironmentClient()
 
         public bool IsInLspEditorContext()
         {
-            return IsLiveShareGuest() || IsCloudEnvironmentClient();
+            var featureFlagService = _workspace.Services.GetRequiredService<IExperimentationService>();
+            var isInLspContext = IsLiveShareGuest() || IsCloudEnvironmentClient() || featureFlagService.IsExperimentEnabled(LspEditorFeatureFlagName);
+            return isInLspContext;
```

Comment:
is any of this expensive.  should we compute and cache once?

Reply:
Feature flags should be cached by the underlying service - but IsLiveShareGuest cannot be cached as a user can switch to join a liveshare session in the same VS instance.

Reply:
> Feature flags should be cached by the underlying service

Which underlying service? I've seen perf bugs in VS because the VS service didn't do caching itself.

Reply:
@jasonmalinowski they didn't use to cache them, but they should be now https://devdiv.visualstudio.com/DevDiv/_workitems/edit/1128651